cea/technologies/thermal_network/thermal_network_costs.py

```diff
@@ -0,0 +1,563 @@
+    # read in total pressure loss in kW
+    deltaP_kW = pd.read_csv(network_info.locator.get_ploss('', network_type))
+    deltaP_kW = deltaP_kW['pressure_loss_total_kW'].sum()
+
+    Opex_var = deltaP_kW * 1000 * network_info.prices.ELEC_PRICE
+
+    if network_info.config.thermal_network.network_type == 'DH':
+        deltaPmax = np.max(network_info.network_features.DeltaP_DHN)
+    else:
+        deltaPmax = np.max(network_info.network_features.DeltaP_DCN)
+    Capex_a, Opex_fixed = pumps.calc_Cinv_pump(deltaPmax, mdotnMax_kgpers, PUMP_ETA, network_info.config,
+                                               network_info.locator, 'PU1')  # investment of Machinery
+    Opex_a_tot = Opex_var + Opex_fixed
+
+    return Capex_a, Opex_a_tot
+
+
+def calc_Ctot_cooling_plants(network_info):
+    """
+    Calculates costs of centralized cooling plants (chillers and cooling towers).
+    :param network_info: an object storing information of the current network
+    :return:
+    """
+
+    # read in plant heat requirement
+    plant_heat_hourly_kWh = pd.read_csv(
+        network_info.locator.get_optimization_network_layout_plant_heat_requirement_file(
+            network_info.network_type, network_info.config.thermal_network_optimization.network_name))
+    # read in number of plants
+    number_of_plants = len(plant_heat_hourly_kWh.columns)
+
+    plant_heat_original_kWh = plant_heat_hourly_kWh.copy()
+    plant_heat_peak_kW_list = plant_heat_hourly_kWh.abs().max(axis=0).values  # calculate peak demand
+    plant_heat_sum_kWh_list = plant_heat_hourly_kWh.abs().sum().values  # calculate aggregated demand
+
+    Opex_var_chiller = 0.0
+    Opex_var_CT = 0.0
+    Capex_a_chiller = 0.0
+    Opex_a_chiller = 0.0
+    Capex_a_CT = 0.0
+    Opex_a_CT = 0.0
+    # calculate cost of chiller heat production and chiller capex and opex
+    for plant_number in range(number_of_plants):  # iterate through all plants
+        if number_of_plants > 1:
+            plant_heat_peak_kW = plant_heat_peak_kW_list[plant_number]
+        else:
+            plant_heat_peak_kW = plant_heat_peak_kW_list[0]
+        plant_heat_yearly_kWh = plant_heat_sum_kWh_list[plant_number]
+        print 'Annual plant heat production:', round(plant_heat_yearly_kWh, 0), '[kWh]'
+        Capex_chiller = 0
+        Opex_fixed_chiller = 0
+        Capex_CT = 0
+        Opex_fixed_CT = 0
+        if plant_heat_peak_kW > 0:  # we have non 0 demand
+            peak_demand_W = plant_heat_peak_kW * 1000  # convert to W
+            print 'Calculating cost of heat production at plant number: ', plant_number
+            if network_info.config.thermal_network_optimization.yearly_cost_calculations:
+                # calculates operation costs with yearly approximation
+                COP_plant = VCCModel.calc_VCC_COP(network_info.config,
+                                                  network_info.full_cooling_systems,
```

Comment:
why not using line 164 to get the exact supplied systems?

Reply:
My reasoning was that since this is the yearly calculation we just take the inputs, and the funciton in the hourly calculation evaluates every hour. But I could adapt the 'find_non_zero_demand_systems' hourly function from line 164 to evaluate the aggregated yearly values. Then I could use the function here too.

Reply:
I am adding this function in the next commit